### ISSUE TITLE
electron.screen.getAllDisplays( ) keeps returning initial screen available space

### ISSUE THREAD

**Issue by pouyakary** [electron/electron#18622]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
  v5.0.2
* **Operating System:**
  macOS 10.14.2 
* **Last Known Working Electron version:**
  `-`

### Expected Behavior
The `screen` module of the `electron` package has a method called `getAllDisplays` which should return the size of the screen as well as the available space for the app. In Mac, the dock size is dynamic and therefore by changing the size of the dock, the available space should change as well.

### Actual Behavior

As you can see, regardless of the dock size the system returns a constant `701` number

<img width="1280" alt="Screen Shot 1398-03-14 at 1 14 44 PM" src="https://user-images.githubusercontent.com/2157285/58865241-758a6d00-86cb-11e9-8f45-8c036f2725cf.png">

<img width="1280" alt="Screen Shot 1398-03-14 at 1 15 00 PM" src="https://user-images.githubusercontent.com/2157285/58865243-76230380-86cb-11e9-82c3-6dcdebf34115.png">

**Comment by codebytere** [comment]
@pmkary i can dig into this a bit: but we don't presently change anything as it's filtered through Chromium's screen capture code, and it does not seem that code currently takes occlusion by dock into account when making these calculations.

**Comment by pmkary** [comment]
@codebytere &mdash; Shelly, hi! thanks a lot for looking into it.

For multi-window software (say financial terminals) of which window management is internally handled, it is very important since we get windows under the dock by API fault all the time. So if was possible to have the API use native OS calls instead of chromium, it could be marvelous.

**Comment by bpasero** [comment]
@deepak1556 @codebytere I wonder if https://github.com/microsoft/vscode/issues/74872 is related, which is a regression we see after updating to Electron 4.2.3. It looks like a Window cannot become larger than the primary display bounds.

**Comment by MarshallOfSound** [comment]
@bpasero I don't believe that these issues are related (at least at a glance).  Could you try (wherever you are restoring the position), calling `setPosition` / `setBounds` twice in a row instead of once? Just to help me check a theory.

**Comment by pmkary** [comment]
I gathered a little bit more of info into this. Changing the position of the dock (`bottom`/`left`/`right`) also does not make any difference too as well as hiding the menubar. So for it seems that the chromium API computes the available area only once and then keeps returning those numbers regardless of the changes.

**Comment by pmkary** [comment]
@MarshallOfSound @bpasero &bullet; I think that Benjamin is right about these issues being connected. We encountered the same issue as vscode's 74872 on a multi-window system where we had to manually align windows on multiple monitors and changes in monitors did break the system. Our windows ended in strange places like under the mac's menubar. At that time we used OpenFin (an electron fork) so we thought it's OpenFin's API fault, (it kept returning initial monitor/display info). But now that I see this within API of electron and since our issues are pretty much the same, I'm 100% certain the two issues are connected.

**Comment by bpasero** [comment]
@pmkary does your issue reproduce in Electron 4 and not in 3?

**Comment by pmkary** [comment]
@bpasero &mdash; We used OpenFin which didn't specify its version mapping to electron, but I can test it on an electron playground. So I'll be testing it

**Comment by bpasero** [comment]
@MarshallOfSound I can confirm that the following workaround makes the issue go away:
* create the window (with the proper size)
* call `win.setBounds()` after that with the same values

Is this a known issue? Does it impact other OS than macOS? Should we go with this workaround? I think this should also be conditional when more than one monitor is used as I do not think this issue reproduces on a single monitor.

//cc @deepak1556

**Comment by bpasero** [comment]
I pushed this workaround for VS Code.

**Comment by bpasero** [comment]
> @bpasero I don't believe that these issues are related (at least at a glance). Could you try (wherever you are restoring the position), calling setPosition / setBounds twice in a row instead of once? Just to help me check a theory.

@MarshallOfSound this is actually causing yet another issue (https://github.com/microsoft/vscode/issues/75830) when using macOS native tabs. Here is what happens (macOS native tabs enabled in VSCode):
* you start with one window and maximize it
* you open a second window
* this adds a new tab to the native tab bar (provided you have configured to always open as tab when opening new windows)
* I then call setBounds() with a value as workaround
* the window now resizes to that value, even though it should not resize given the first window dimensions are already correct

I will have to add another workaround to never call `setBounds()` when there is more than one window opened, which is odd.

**Comment by carterbs** [comment]
+1 that this issue is causing us grief. We want to position a window at the bottom of the monitor. My expectation is that workArea and workAreaSize will change as the dock's position changes.

**Comment by electron-triage** [comment]
The Electron version reported on this issue is no longer supported. See our [supported versions documentation](https://www.electronjs.org/docs/tutorial/support#supported-versions).

If this is still reproducible on a supported version, please open a new issue with any other new information that a maintainer should know.

Thank you for taking the time to report this issue and helping to make Electron better! Your help is appreciated.

**Comment by bpasero** [comment]
This issue should be reopened, even with latest Electron 28, I need to call `setBounds` on windows to properly restore the size in a multi-monitor setup when resolutions are different.
